code/modules/clothing/accessories/accessory.dm

```diff
@@ -319,3 +319,51 @@
 		source_vest = null
 	qdel(src)
 
+
+/obj/item/clothing/accessory/rad_patch
+	name = "radiation detection patch"
+	desc = "A paper patch that you can attach to your clothing. Changes color to black when it absorbs over a certain amount of radiation"
+	icon_state = "rad_patch"
+	var/rad_absorbed = 0
+	var/rad_threshold = 45
+	var/triggered = FALSE
+	var/event_key
+	autoignition_temperature = AUTOIGNITION_PAPER
+	fire_fuel = 1
+	w_class = W_CLASS_TINY
+	w_type = RECYK_WOOD
+
+/obj/item/clothing/accessory/rad_patch/proc/check_rads(list/arguments)
+	if(triggered)
+		return
+	var/mob/user = arguments["user"]
+	var/rads = arguments["rads"]
+	rad_absorbed += rads
+
+	if(!triggered && rad_absorbed > rad_threshold)
```

Comment:
The `!triggered` check is redundant. Also it should remove the event listener in here, unless you have a particular reason for it continuing to run the proc after the thing is already triggered.